sktree/datasets/hyppo.py

```diff
@@ -43,3 +45,297 @@ def make_quadratic_classification(n_samples: int, n_features: int, noise=False,
     v = np.vstack([np.zeros((n1, 1)), np.ones((n2, 1))])
     x = np.vstack((x, y))
     return x, v
+
+
+def make_trunk_classification(
+    n_samples,
+    n_dim=10,
+    m_factor: int = -1,
+    rho: int = 0,
+    band_type: str = "ma",
+    return_params: bool = False,
+    mix: int = 0,
+    seed=None,
+):
+    """Generate trunk dataset.
+
+    For each dimension in the first distribution, there is a mean of :math:`1 / d`, where
+    ``d`` is the dimensionality. The covariance is the identity matrix.
+    The second distribution has a mean vector that is the negative of the first.
+    As ``d`` increases, the two distributions become closer and closer.
+
+    See full details in :footcite:`trunk1982`.
+
+    Instead of the identity covariance matrix, one can implement a banded covariance matrix
+    that follows :footcite:`Bickel_2008`.
+
+    Parameters
+    ----------
+    n_samples : int
+        Number of sample to generate.
+    n_dim : int, optional
+        The dimensionality of the dataset and the number of
+        unique labels, by default 10.
+    m_factor : int, optional
+        The multiplicative factor to apply to the mean-vector of the first
+        distribution to obtain the mean-vector of the second distribution.
+        By default -1.
+    rho : float, optional
+        The covariance value of the bands. By default 0 indicating, an identity matrix is used.
+    band_type : str
+        The band type to use. For details, see Example 1 and 2 in :footcite:`Bickel_2008`.
+        Either 'ma', or 'ar'.
+    return_params : bool, optional
+        Whether or not to return the distribution parameters of the classes normal distributions.
+    mix : float, optional
+        Whether or not to mix the Gaussians. Should be a value between 0 and 1.
+    seed : int, optional
+        Random seed, by default None.
+
+    Returns
+    -------
+    X : np.ndarray of shape (n_samples, n_dim), dtype=np.float64
+        Trunk dataset as a dense array.
+    y : np.ndarray of shape (n_samples,), dtype=np.intp
+        Labels of the dataset.
+    means : list of ArrayLike of shape (n_dim,), dtype=np.float64
+        The mean vector for each class starting with class 0.
+        Returned if ``return_params`` is True.
+    covs : list of ArrayLike of shape (n_dim, n_dim), dtype=np.float64
+        The covariance for each class. Returned if ``return_params`` is True.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    rng = np.random.default_rng(seed=seed)
+
+    mu_1 = np.array([1 / np.sqrt(i) for i in range(1, n_dim + 1)])
```

Comment:
Aren't these `mu`'s reversed from what we defined on board? F should be `mu` and G should be `-mu`. @sampan501 @adam2392

Reply:
I think it should be fine since it is symmetrical.

Reply:
I don't think we should add extra confusion in it.

Reply:
I personally don't feel strongly either way

Reply:
ok